experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -447,29 +450,34 @@ export type ObjectNodeTyped<TSchema extends ObjectNodeSchema> = ObjectNode &
  *
  * @alpha
  */
-export type ObjectNodeFields<TFields extends RestrictiveReadonlyRecord<string, TreeFieldSchema>> = {
-	// boxed fields (TODO: maybe remove these when same as non-boxed version?)
-	readonly [key in keyof TFields as `boxed${Capitalize<key & string>}`]: TypedField<TFields[key]>;
-} & {
-	// Add getter only (make property readonly) when the field is **not** of a kind that has a logical set operation.
-	// If we could map to getters and setters separately, we would preferably do that, but we can't.
-	// See https://github.com/microsoft/TypeScript/issues/43826 for more details on this limitation.
-	readonly [key in keyof TFields as TFields[key]["kind"] extends AssignableFieldKinds
-		? never
-		: key]: UnboxField<TFields[key]>;
-} & {
-	// Add setter (make property writable) when the field is of a kind that has a logical set operation.
-	// If we could map to getters and setters separately, we would preferably do that, but we can't.
-	// See https://github.com/microsoft/TypeScript/issues/43826 for more details on this limitation.
-	-readonly [key in keyof TFields as TFields[key]["kind"] extends AssignableFieldKinds
-		? key
-		: never]: UnboxField<TFields[key]>;
-} & {
-	// Setter method (when the field is of a kind that has a logical set operation).
-	readonly [key in keyof TFields as TFields[key]["kind"] extends AssignableFieldKinds
-		? `set${Capitalize<key & string>}`
-		: never]: (content: FlexibleFieldContent<TFields[key]>) => void;
-};
+export type ObjectNodeFields<TFields extends RestrictiveReadonlyRecord<string, TreeFieldSchema>> =
+	FlattenKeys<
```

Comment:
this makes intelisense cleaner, and is does not impact type safty as far as I know.

Reply:
It would probably be worth noting this (and any other idiosyncrasies) in a `@privateRemarks` comment block for future reference. These types are hard to reason about, especially for folks not well versed in this code. So the more we can document about "odd" choices we've made in the code the better.

Reply:
FlattenKeys has a doc comment explaining it. It exists as a type with a name for the purpose of being documentable. I don't think we also need to document every place its used.